casper/src/main/scala/coop/rchain/casper/Casper.scala

```diff
@@ -164,8 +184,10 @@ sealed abstract class MultiParentCasperInstances {
         } yield ()
 
       def estimator: F[IndexedSeq[BlockMessage]] =
-        Capture[F].capture {
-          Estimator.tips(_blockDag.get, genesis)
+        BlockStore[F].asMap() >>= { internalMap: Map[BlockHash, BlockMessage] =>
+          Capture[F].capture {
```

Comment:
If you're heavily using `cats` and `cats-effect` I would recommend migrate from `Capture` to `Sync`.

Reply:
Agreed. I just don't want to introduce new complexity and mix it with a change that can be done separately.

Reply:
This would require a lot of changes, so a separate PR is a must